src/nvim/eval.c

```diff
@@ -14456,11 +14456,11 @@ static void f_setreg(typval_T *argvars, typval_T *rettv)
   char_u      *strregname;
   char_u      *stropt;
   bool append = false;
-  char_u yank_type;
+  MotionType yank_type;
   long block_len;
 
   block_len = -1;
-  yank_type = MAUTO;
+  yank_type = kUnknownMotion;
```

Comment:
There is no place for “Unknown” here. Prose is fine with “when register is linewise”, it is not code. Also it is still possible to have common prefix and Wise: kMTUnknown, kMTCharacterWise, …